diff -u -r orig-libcec-2fc92b5/src/libcec/cmake/git-rev.sh libcec-2fc92b5/src/libcec/cmake/git-rev.sh
--- orig-libcec-2fc92b5/src/libcec/cmake/git-rev.sh     2021-10-14 18:01:38.879639648 +0100
+++ libcec-2fc92b5/src/libcec/cmake/git-rev.sh  2021-10-14 18:02:38.510544800 +0100
@@ -2,7 +2,7 @@

 ## cmake doesn't read the variable when it doesn't end with a newline, and I haven't figured out how to have it add a newline directly...
 if git rev-parse --git-dir > /dev/null 2>&1; then
-  last_tag=`git describe --tags --abbrev=0`
+  last_tag=`git describe --always --tags --abbrev=0`
   last_hash=`git --no-pager log --abbrev=7 -n 1 --pretty=format:"%h"`
   commits_since_tag=`git log ${last_tag}..HEAD --oneline | wc -l`
   git_dirty=`git diff HEAD | wc -l`
